source/PlayerInfo.cpp

```diff
@@ -49,6 +49,26 @@ PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 
 using namespace std;
 
+namespace {
+	// Move the flagship to the start of your list of ships. It does not make sense
+	// that the flagship would change if you are reunited with a different ship that
+	// was higher up the list.
+	// This function should not be used while other functions have direct references
+	// to the shared_pointer elements in this list.
+	void MoveFlagshipBegin(vector<shared_ptr<Ship>> &ships, const shared_ptr<Ship> &flagship)
+	{
+		if(!flagship)
+			return;
+
+		auto it = find(ships.begin(), ships.end(), flagship);
+		if(it != ships.begin() && it != ships.end())
+		{
+			move_backward(ships.begin(), prev(it), it);
```

Comment:
```suggestion
			move_backward(ships.begin(), it, next(it));
```

The range doesn't include the last element. As such the end is `it`. And the third argument is the new end, i.e. one past the actual end so we need to increment it by one.

Reply:
Sorry, I should have read the documentation better.